fix(snack-bar): add content fade in animation

crisbeto opened the pull request:
Adds a subtle fade in animation to the snack bar content.

Fixes #7502.

josephperrott commented on src/lib/snack-bar/simple-snack-bar.ts:
```diff
@@ -23,7 +24,15 @@ import {MAT_SNACK_BAR_DATA} from './snack-bar-config';
   changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger('contentFade', [
+      state('void', style({opacity: '0'})),
```
I think you can just use `:enter` instead of having to manage states.

josephperrott commented on src/lib/snack-bar/simple-snack-bar.ts:
```diff
@@ -23,7 +24,15 @@ import {MAT_SNACK_BAR_DATA} from './snack-bar-config';
+    trigger('contentFade', [
+      state('void', style({opacity: '0'})),
+      state('visible', style({opacity: '1'})),
+      transition('void => visible', animate('350ms linear'))
```
Should we use the animations from core?

```
`${AnimationCurves.STANDARD_CURVE} ${AnimationDurations.ENTERING}`
```

willshowell replied:
Does that work now with AOT? I thought string interpolation in animation [broke it](https://github.com/angular/material2/blob/master/src/lib/snack-bar/snack-bar-container.ts#L41-L42).

crisbeto replied:
That comment is pretty old. It seems to work with AoT now.

crisbeto commented:
Addressed the feedback @josephperrott, PTAL. One note: I'm not too sure about using the `COMPLEX` animation duration, but all of the others were too short.

josephperrott commented:
Yeah, what we likely would like to do is delay the start of the animation by ~100ms so that it times better, but I think that increasing the overall length will work just as well.

kara commented:
@crisbeto Linter?

crisbeto commented:
Fixed.
